### ISSUE TITLE
False positive for numpy 2.4.0+

### ISSUE THREAD

**Issue by owillebo** [pylint-dev/pylint#10806]
### Bug description

```python
pylint can't find the eps attribute for np.finfo(float).eps with numpy 2.4.0+
```

### Configuration

```ini
default
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
test.py:3:4: E1101: Instance of 'finfo' has no 'eps' member (no-member)
```

### Expected behavior

No issue

### Pylint version

```shell
pylint 4.0.4
astroid 4.0.3
Python 3.14.2 (tags/v3.14.2:df79316, Dec  5 2025, 17:18:21) [MSC v.1944 64 bit (AMD64)]
```

### OS / Environment

```
RMDIR /S /Q .venv
ERASE *.py
py -3.14-64 -m venv .venv
CALL .venv\Scripts\activate
@ECHO ON
python -m pip install -U pip
python -m pip install pylint numpy
python --version
python -m pip freeze
ECHO """Triggers pylint issue."""> test.py
ECHO import numpy as np>> test.py
ECHO _ = np.finfo(float).eps>> test.py
REM ~ pylint can't find the eps attribyte for numpy 2.4.0+
pylint test.py
python -m pip install numpy==2.3.5
python --version
python -m pip freeze
REM ~ pylint has no problems with numpy 2.3.5
pylint test.py
```

```
C:\Temp\mypy>RMDIR /S /Q .venv 

C:\Temp\mypy>ERASE *.py 

C:\Temp\mypy>py -3.14-64 -m venv .venv 

C:\Temp\mypy>CALL .venv\Scripts\activate 

(.venv) C:\Temp\mypy>python -m pip install -U pip 
Requirement already satisfied: pip in c:\temp\mypy\.venv\lib\site-packages (25.3)

(.venv) C:\Temp\mypy>python -m pip install pylint numpy 
Collecting pylint
  Using cached pylint-4.0.4-py3-none-any.whl.metadata (12 kB)
Collecting numpy
  Using cached numpy-2.4.0-cp314-cp314-win_amd64.whl.metadata (6.6 kB)
Collecting astroid<=4.1.dev0,>=4.0.2 (from pylint)
  Using cached astroid-4.0.3-py3-none-any.whl.metadata (4.4 kB)
Collecting colorama>=0.4.5 (from pylint)
  Using cached colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting dill>=0.3.6 (from pylint)
  Using cached dill-0.4.0-py3-none-any.whl.metadata (10 kB)
Collecting isort!=5.13,<8,>=5 (from pylint)
  Using cached isort-7.0.0-py3-none-any.whl.metadata (11 kB)
Collecting mccabe<0.8,>=0.6 (from pylint)
  Using cached mccabe-0.7.0-py2.py3-none-any.whl.metadata (5.0 kB)
Collecting platformdirs>=2.2 (from pylint)
  Using cached platformdirs-4.5.1-py3-none-any.whl.metadata (12 kB)
Collecting tomlkit>=0.10.1 (from pylint)
  Using cached tomlkit-0.13.3-py3-none-any.whl.metadata (2.8 kB)
Using cached pylint-4.0.4-py3-none-any.whl (536 kB)
Using cached astroid-4.0.3-py3-none-any.whl (276 kB)
Using cached isort-7.0.0-py3-none-any.whl (94 kB)
Using cached mccabe-0.7.0-py2.py3-none-any.whl (7.3 kB)
Using cached numpy-2.4.0-cp314-cp314-win_amd64.whl (12.4 MB)
Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Using cached dill-0.4.0-py3-none-any.whl (119 kB)
Using cached platformdirs-4.5.1-py3-none-any.whl (18 kB)
Using cached tomlkit-0.13.3-py3-none-any.whl (38 kB)
Installing collected packages: tomlkit, platformdirs, numpy, mccabe, isort, dill, colorama, astroid, pylint

Successfully installed astroid-4.0.3 colorama-0.4.6 dill-0.4.0 isort-7.0.0 mccabe-0.7.0 numpy-2.4.0 platformdirs-4.5.1 pylint-4.0.4 tomlkit-0.13.3

(.venv) C:\Temp\mypy>python --version 
Python 3.14.2

(.venv) C:\Temp\mypy>python -m pip freeze 
astroid==4.0.3
colorama==0.4.6
dill==0.4.0
isort==7.0.0
mccabe==0.7.0
numpy==2.4.0
platformdirs==4.5.1
pylint==4.0.4
tomlkit==0.13.3

(.venv) C:\Temp\mypy>ECHO """Triggers pylint issue.""" 1>test.py 

(.venv) C:\Temp\mypy>ECHO import numpy as np 1>>test.py 

(.venv) C:\Temp\mypy>ECHO _ = np.finfo(float).eps 1>>test.py 

(.venv) C:\Temp\mypy>REM ~ pylint can't find the eps attribyte for numpy 2.4.0+ 

(.venv) C:\Temp\mypy>pylint test.py 
************* Module test
test.py:3:4: E1101: Instance of 'finfo' has no 'eps' member (no-member)

--------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 10.00/10, -10.00)


(.venv) C:\Temp\mypy>python -m pip install numpy==2.3.5 
Collecting numpy==2.3.5
  Using cached numpy-2.3.5-cp314-cp314-win_amd64.whl.metadata (60 kB)
Using cached numpy-2.3.5-cp314-cp314-win_amd64.whl (12.9 MB)
Installing collected packages: numpy
  Attempting uninstall: numpy
    Found existing installation: numpy 2.4.0
    Uninstalling numpy-2.4.0:
      Successfully uninstalled numpy-2.4.0
Successfully installed numpy-2.3.5

(.venv) C:\Temp\mypy>python --version 
Python 3.14.2

(.venv) C:\Temp\mypy>python -m pip freeze 
astroid==4.0.3
colorama==0.4.6
dill==0.4.0
isort==7.0.0
mccabe==0.7.0
numpy==2.3.5
platformdirs==4.5.1
pylint==4.0.4
tomlkit==0.13.3

(.venv) C:\Temp\mypy>REM ~ pylint has no problems with numpy 2.3.5 

(.venv) C:\Temp\mypy>pylint test.py 

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

### Additional dependencies

```python
astroid==4.0.3
colorama==0.4.6
dill==0.4.0
isort==7.0.0
mccabe==0.7.0
numpy==2.4.0
platformdirs==4.5.1
pylint==4.0.4
tomlkit==0.13.3
```
